sound/soc/sof/ipc4-pcm.c

```diff
@@ -106,7 +211,29 @@ static int sof_ipc4_trigger_pipelines(struct snd_soc_component *component,
 		pipeline->state = state;
 	}
 
-	return ret;
+	if (ret < 0)
+		return ret;
+
+	if (state == SOF_IPC4_PIPE_RUNNING) {
+		u32 abi_version;
+		u32 abi_offset;
+
+		abi_offset = offsetof(struct sof_ipc4_fw_registers, abi_ver);
+		sof_mailbox_read(sdev, sdev->fw_info_box.offset + abi_offset, &abi_version,
+				 sizeof(abi_version));
+
+		if (abi_version != SOF_IPC4_FW_REGS_ABI_VER) {
+			dev_warn(sdev->dev, "unsupported abi version %d for time stamp\n",
```

Comment:
if you return an error it should be dev_err, no?

Reply:
sure, use dev_error